src/ari_sxn_simbeamline/xrt_sim.py

```diff
@@ -0,0 +1,131 @@
+
+def buildBeamline(eV, sigma_eV, Nrays, HsltSz=4, VsltSz=4):
+    # build a properly aligned beamline
+
+    # global coordinate system:
+    #   x transverse to source beam
+    #   y along source beam propgation direction
+    #   z is up
+    #   position of optical elements are given in the global coordinate system
+    #   rotations are applied in order yaw, roll, pitch
+    #   rotations of optical elements are done in the local coordinate system
+    #      in the default order yaw, roll, pitch
+
+    d_Src2Scrn = 28373  # mm
+    d_Src2Slit = 28000  # mm
+    d_Src2Mirr = 27850  # mm
+
+    mat_mirror = gold
+    pit_mirror = 0.0000218  # rad
+    # Build beamline.  Set azimuth parameter in beamline to 0.  Then
+    # global coordinate system and beamline coordinate system are identical
+    # and they are 'SHADOW' like (right hand coordinate system)
+    bl = raycing.BeamLine(azimuth=0.0, height=0.0, alignE=0)
+
+    # create geometric source
+    bl.source = rs.GeometricSource(
+        bl=bl,
+        name='test1',
+        center=(0, 0, 0),
+        nrays=Nrays,
+        distx='normal', dx=0.030,
+        disty=None, dy=0,
+        distz='normal', dz=0.03,
+        distxprime='normal', dxprime=0.0001,
+        distzprime='normal', dzprime=0.0001,
+        distE='normal', energies=(eV, sigma_eV),
+        polarization='horizontal',
+        filamentBeam=False,
+        uniformRayDensity=False
+    )
+
+    bl.pm = roe.OE(bl=bl, name='PM', center=[0, d_Src2Mirr, 0],
+                   yaw=0, roll=0, pitch=pit_mirror, material=mat_mirror
+                   )
+
+    bl.xslt = raa.RectangularAperture(bl=bl, name='Exit slit',
+                                      center=np.array([0, d_Src2Slit, 0]), x='auto',
+                                      kind=['left', 'right', 'bottom', 'top'],
+                                      opening=[-HsltSz / 2, HsltSz / 2,
+                                               -VsltSz / 2, VsltSz / 2]
+                                      )
+
+    bl.scrn = rsc.Screen(bl=bl,
+                         name='Screen',
+                         center=np.array([0, d_Src2Scrn, 0]),
+                         x=np.array([1, 0, 0]),
+                         z=np.array([0, 0, 1])
+                         )
+
+    return bl
+
+
+# set directions to raytrace beamline
+def run_process0(bl):
+    beamSource = bl.sources[0].shine()
+    beamPMgl, beamPMloc = bl.pm.reflect(beamSource)
+    beamXslt = bl.xslt.propagate(beamPMgl)
+    beamScrn = bl.scrn.expose(beamPMgl)
+    #
+    # beamXslt = bl.xslt.propagate(beamSource)
+    # beamScrn = bl.scrn.expose(beamSource)
+
+    outdict = {
+        'beamSource': beamSource,
+        'beamPMloc': beamPMloc,
+        'beamXslt': beamXslt,
+        'beamScrn': beamScrn
+    }
+    return outdict
+
+Nrays = 1e4
+eV = 860  # eV
+sigma_eV = 0.05  # eV
+
+bl = buildBeamline(eV, sigma_eV, Nrays)
+beamSource = bl.sources[0].shine()
```

Comment:
It appears these are just used to pass into the dictionary below, can we just directly add these lines in the dict definition? or perhaps look at if we need the beam_dict object?